frontend/model/notifications/utils.js

```diff
@@ -121,3 +120,11 @@ export function isOlder (notification: Notification): boolean {
 export function maxAge (notification: Notification): number {
   return notification.read ? MAX_AGE_READ : MAX_AGE_UNREAD
 }
+
+export function makeNotificationHash (notification: Object): string {
+  const sorted = Object.keys(notification).sort().reduce((acc, curKey) => {
+    acc.push([curKey, notification[curKey]])
+    return acc
+  }, [])
```

Comment:
Please create an algorithm that recursively checks whether the value (`notification[curKey]`) is an object, and if so performs the same reduction on the object (to an array). Some notification data might include objects, as well as nested objects, and if we're going to properly create the hash for the outer object, we should do the same for any inner objects.